modules/tinymce/src/core/main/ts/caret/LineWalker.ts

```diff
@@ -96,16 +101,18 @@ const positionsUntil = (direction: VDirection, root: Node, predicateFn: RectPred
   let walkFn: (caretPosition: CaretPosition | null) => CaretPosition | null;
   let isBelowFn: PosPredicate;
   let isAboveFn: PosPredicate;
-  let caretPosition: CaretPosition;
+  let caretPosition: CaretPosition | null;
   const result: LinePosClientRect[] = [];
   let line = 0;
 
-  const getClientRect = (caretPosition: CaretPosition) => {
-    if (direction === 1) {
-      return ArrUtils.last(caretPosition.getClientRects());
+  const getClientRect = (caretPosition: CaretPosition): ClientRect => {
+    // TODO: Are we really guaranteed that we're get at least 1 client rect back?
+    // Also should this really be returning the last item for both directions?
+    if (direction === VDirection.Down) {
+      return ArrUtils.last(caretPosition.getClientRects()) as ClientRect;
+    } else {
+      return ArrUtils.last(caretPosition.getClientRects()) as ClientRect;
     }
```

Comment:
This is almost definitely a bug... however the git history shows it's been like that since at least 2015. Should we just remove the branching here?

Reply:
I would think the intent is to return the last for one and first for the other (not sure which is which). Maybe there's been a weird lurking bug since 2015?

I believe at least one client rect is guaranteed - for a range `getClientRects` aggregates all client rects of elements in the range. I guess if it's not connected to the body it might be zero.

Reply:
As discussed today, this logic is only used when for the `isAboveFn` so using `last` appears to make sense there. As such we're just going to remove the `direction` check and rename the function.